sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/Create.java

```diff
@@ -633,11 +650,48 @@ public T apply(TimestampedValue<T> input) {
                 + " Based on their values, they do not all default to the same Coder.");
       }
     }
+    if (coder.isPresent()) {
+      return coder.get();
+    }
 
-    if (!coder.isPresent()) {
-      throw new CannotProvideCoderException(
-          "Unable to infer a coder. Please register " + "a coder for ");
+    throw new CannotProvideCoderException("Cannot provide coder for elements of "
+        + Create.class.getSimpleName()
+        + ":"
+        + " For their common class, no coder could be provided."
+        + " Based on their values, no coder could be inferred.");
+  }
+
+  /**
+   * Attempt to infer the type for some very common Apache Beam parameterized types.
+   *
+   * <p>TODO: Instead, build a TypeDescriptor so that the {@link CoderRegistry} is invoked
```

Comment:
Actually this is basically as good as status quo and we can make any changes backwards compatibly. And the feature is not really type safe to begin with. So scrap this comment.